src/java/org/apache/cassandra/config/DataRateSpec.java

```diff
@@ -0,0 +1,369 @@
+    {
+        return Ints.saturatedCast(toMegabitsPerSecond());
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(unit.toKibibytesPerSecond(quantity));
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+
+        if (!(obj instanceof DataRateSpec))
+            return false;
+
+        DataRateSpec other = (DataRateSpec) obj;
+        if (unit == other.unit)
+            return quantity == other.quantity;
+
+        // Due to overflows we can only guarantee that the 2 data rates are equal if we get the same results
+        // doing the conversion in both directions.
+        return unit.convert(other.quantity, other.unit) == quantity && other.unit.convert(quantity, unit) == other.quantity;
+    }
+
+    @Override
+    public String toString()
+    {
+        return quantity + unit.symbol;
+    }
+
+    public enum DataRateUnit
+    {
+        BYTES_PER_SECOND("B/s")
+        {
+            public long toBytesPerSecond(long d)
+            {
+                return d;
+            }
+
+            public long toKibibytesPerSecond(long d)
+            {
+                return d / 1024;
+            }
+
+            public long toMebibytesPerSecond(long d)
+            {
+                return d / (1024 * 1024);
+            }
+
+            public long toMegabitsPerSecond(long d) { return d / 125000; }
+
+            public long convert(long source, DataRateUnit sourceUnit)
+            {
+                return sourceUnit.toBytesPerSecond(source);
+            }
+        },
+        KIBIBYTES_PER_SECOND("KiB/s")
+        {
+            public long toBytesPerSecond(long d)
+            {
+                return x(d, 1024, MAX / 1024);
+            }
+
+            public long toKibibytesPerSecond(long d)
+            {
+                return d;
+            }
+
+            public long toMebibytesPerSecond(long d)
+            {
+                return d / 1024;
+            }
+
+            public long toMegabitsPerSecond(long d)
+            {
+                return d / 122;
+            }
+
+            public long convert(long source, DataRateUnit sourceUnit)
+            {
+                return sourceUnit.toKibibytesPerSecond(source);
+            }
+        },
+        MEBIBYTES_PER_SECOND("MiB/s")
+        {
+            public long toBytesPerSecond(long d)
+            {
+                return x(d, 1024 * 1024, MAX / (1024 * 1024));
+            }
+
+            public long toKibibytesPerSecond(long d)
+            {
+                return x(d, 1024, MAX / (1024));
+            }
+
+            public long toMebibytesPerSecond(long d)
+            {
+                return d;
+            }
+
+            public long toMegabitsPerSecond(long d)
+            {
+                if ((double)d > MAX / (8.38861))
+                    return Long.MAX_VALUE;
+                return Math.round(d * 8.38861);
```

Comment:
nit: Might be worth pulling a constant out of this like `MEGABITS_PER_MEBIBYTE`

Reply:
Done, plus changed the usage of `Long.MAX_VALUE` with `MAX`